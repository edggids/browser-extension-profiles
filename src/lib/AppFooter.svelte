<script lang="ts">
export let style = '';
</script>

<footer class="container transition-color text-current-darker flex justify-between pb-6.5" {style} style:--color="var(--color-current-darker)">
    <slot></slot>
</footer>
