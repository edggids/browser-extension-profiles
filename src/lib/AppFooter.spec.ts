import {cleanup, render} from '@testing-library/svelte';
import {describe, it, expect, afterEach} from 'vitest';
import AppFooter from "./AppFooter.svelte";

describe('AppFooter component', () => {
  it('mounts', () => {
    const { container } = render(AppFooter);
    expect( container ).toMatchSnapshot();
  });

  it('renders with custom style', () => {
    const { container } = render(AppFooter, { props: { style: 'background-color: red;' } });
    const footer = container.querySelector('footer');

    expect( footer.style.backgroundColor ).toBe('red');
    expect( footer ).toMatchSnapshot();
  });
});
