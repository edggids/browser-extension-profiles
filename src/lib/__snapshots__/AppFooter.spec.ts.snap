// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppFooter component > mounts 1`] = `
<body>
  <div>
    <footer
      class="container transition-color text-current-darker flex justify-between pb-6.5"
      style="--color: var(--color-current-darker);"
    />
  </div>
</body>
`;

exports[`AppFooter component > renders with custom style 1`] = `
<footer
  class="container transition-color text-current-darker flex justify-between pb-6.5"
  style="background-color: red; --color: var(--color-current-darker);"
/>
`;
